hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -1788,6 +1788,34 @@ public QuotaUsage getQuotaUsage(Path f) throws IOException {
     return getContentSummary(f);
   }
 
+  /**
+   * Set quota for the given {@link Path}.
+   *
+   * @param src the target path to set quota for
+   * @param namespaceQuota the namespace quota (i.e., # of files/directories) to set
+   * @param storagespaceQuota the storage space quota to set
+   * @throws IOException IO failure
+   */
+  public void setQuota(Path src, final long namespaceQuota,
+      final long storagespaceQuota) throws IOException {
+    throw new UnsupportedOperationException(getClass().getCanonicalName() +
```

Comment:
This is a pattern used in many places.
I think we should have a static method to do this.
I think the easiest way to get the function name is to get the stack trace.

Reply:
Agree. Better to have a static method and make the error message standardized. Not sure what you mean by getting the function name from the stack trace though. I was thinking to just pass the method name to the static helper method.

Do you want to do this cleanup in this JIRA, or address it separately in a follow-up?

Reply:
Let's do just the new two functions in this JIRA so we introduce the function.
Afterwards we can do a separate JIRA to replace all.
To get the method, one could do Thread.currentThread().getStackTrace()[index].getMethodName().
The index sometimes is tricky, but doable.